ocaml/xapi/xapi_vm.ml

```diff
@@ -132,10 +148,17 @@ let set_memory_static_range ~__context ~self ~min ~max =
 		static_max = max;
 	} in
 	Vm_memory_constraints.assert_valid_for_current_context
-		~__context ~constraints;
+		~__context ~vm:self ~constraints;
 	Db.VM.set_memory_static_min ~__context ~self ~value:min;
 	Db.VM.set_memory_static_max ~__context ~self ~value:max;
-	update_memory_overhead ~__context ~vm:self
+	update_memory_overhead ~__context ~vm:self;
+	if Helpers.is_domain_zero ~__context self then
+		set_dom0_memory ~__context ~self ~bytes:max;
+	(* It is allowed to update the memory settings of a running control domain,
+	 * but it needs to be rebooted for the changes to take effect. We signal
+	 * the client to do so. *)
+	if is_control_domain && power_state = `Running then
+		Db.VM.set_requires_reboot ~__context ~self ~value:true
```

Comment:
We have:

* For dom0: https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/create_misc.ml#L345.
* For other VMs: https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/xapi_vm_lifecycle.ml#L521.